### PR TITLE
fix elu/celu/selu gradient being nan on large inputs

### PR DESCRIPTION
**Pull request by teddytennant**
exp is taken over the whole input and the positive half is thrown away, so the gradient is nan past ~88.7. clamping the input to exp fixes it and the forward is unchanged. the tests only ran on [-2, 2].

### CODE REVIEW

File: test/backend/test_ops.py [hunk 1]
```diff
@@ -978,9 +978,11 @@ def test_leaky_relu(self):
   def test_celu(self):
     for val in range(1, 5):
       helper_test_op([(45,65)], lambda x: torch.nn.functional.celu(x,val), lambda x: x.celu(val))
+      helper_test_op([(45,65)], lambda x: torch.nn.functional.celu(x,val), lambda x: x.celu(val), low=300, high=400)
```
Comment (chenyuxyz) [inline test/backend/test_ops.py]: don't need full (45,65) that's a lot of tests

Comment (teddytennant) [reply test/backend/test_ops.py]: dropped to (3,3), and removed the vals check in test_elu.

File: tinygrad/mixin/elementwise.py [hunk 2]
```diff
@@ -995,7 +995,7 @@ def selu(self, alpha=1.67326, gamma=1.0507) -> Self:
     print(Tensor([-3., -2., -1., 0., 1., 2., 3.]).selu().numpy())
     ```
     """
-    return gamma * (self >= 0).where(self, alpha * (self.exp() - 1))
+    return gamma * (self >= 0).where(self, alpha * ((self - self.relu()).exp() - 1))
```
Comment (chenyuxyz) [inline tinygrad/mixin/elementwise.py]: `return gamma * self.elu(alpha)`, also test `0` on this, i think the old code was wrong

Comment (teddytennant) [reply tinygrad/mixin/elementwise.py]: yeah, torch takes the negative branch at 0 so the grad should be gamma*alpha, `>= 0` gave gamma. fixed, and added 0 to test_selu.

also did celu the same way, `alpha * (self / alpha).elu()`.

Comment (chenyuxyz) [reply tinygrad/mixin/elementwise.py]: nice!
